eng/pipelines/templates/steps/build.yml

```diff
@@ -48,12 +48,28 @@ steps:
   - template: /eng/pipelines/templates/steps/set-artifact-packages.yml
     parameters:
       Artifacts: ${{ parameters.Artifacts }}
+  - ${{ if and(eq(variables['System.TeamProject'], 'internal'), in(variables['Build.Reason'], 'IndividualCI', 'BatchedCI')) }}:
+    - task: AzureCLI@2
+      inputs:
+        azureSubscription: 'Azure SDK Artifacts'
+        scriptType: 'bash'
+        scriptLocation: 'inlineScript'
+        inlineScript: |
+          az account show
+          expiry=`date -u -d "45 minutes" '+%Y-%m-%dT%H:%MZ'`
+          sasToken=$(az storage container generate-sas --account-name azuresdkartifacts --name azure-sdk-for-js-rush-cache --permissions dlrw --auth-mode login --as-user --expiry $expiry --https-only -o tsv)
+          echo "##vso[task.setvariable variable=rushBuildCacheCred;]$sasToken"
```

Comment:
I assume you want to make this a secret? Otherwise it will be put as an env variable for all the rest of the steps.

Reply:
See example https://github.com/Azure/azure-sdk-for-js/blob/main/eng/common/TestResources/TestResources-Helpers.ps1#L245

Reply:
updated to secret